### ISSUE TITLE
EUI-64 calculation

### ISSUE THREAD

**Issue by OlegHahm** [RIOT-OS/RIOT#4233]
I just discovered that [RFC4193](https://tools.ietf.org/html/rfc4193#section-3.2.2) describes an algorithm to generate a Pseudo-Random Global ID from either a MAC address or "a suitably unique identifier, local to the node, should be used (e.g., system serial number)". Maybe we can use this algorithm for creating an EUI-64 from the CPUID?

**Comment by authmillenon** [comment]
Looks alright on first side :+1:

**Comment by OlegHahm** [comment]
SHA1 is on its way: #4701.

**Comment by authmillenon** [comment]
Just to be clear, the algorithm described on the RFC still requires the time of date and also describes the creation of a (somewhat) globally unique *IPv6* address, right?

**Comment by OlegHahm** [comment]
Yes, it requires time and date (most of the devices have an RTC and even if not `gettimeofday()` can also be called with xtimer and it doesn't matter if the timestamp is "real"). And it describes how an EUI-64 and an IPv5 address can be obtained.

**Comment by miri64** [comment]
@benpicco @maribu this could be interesting for your rework of `luid`.

**Comment by maribu** [comment]
> @benpicco @maribu this could be interesting for your rework of `luid`.

I'd say the current `luid` implementation is superior in terms of usability, as

- No NTP timestamp is needed for generation in `luid`
- The same board with the same transceiver setup will always end up with the same local EUI in `luid`

**Comment by miri64** [comment]
It _is_ however still problematic when the CPU ID is very similar (see [e.g. last weeks discussion to the users list](https://lists.riot-os.org/pipermail/users/2020-June/001566.html))

**Comment by benpicco** [comment]
Is there a small hash algorithm that produces 64 bit output? Hashing the CPU ID would provide better results than just xor in a loop.

**Comment by maribu** [comment]
IMO this can be closed.

@benpicco: ping :)
